### PR TITLE
UI update. Fixes #32473.

### PR DESCRIPTION
**Pull request by Petethegoat**
Show the number of items required for modification.
Colorize stats that will improve or worsen from a given modification.

I decided to split the cut/bash lines to make the code a bit simpler, but I think this is more readable anyway.

Just PRing for reference, if you prefer to focus on the main functionality first I won't be offended. I can always wait for your PR to merge before working more on this :)

image

### CODE REVIEW

**Comment by lispcoc** [comment]
Seems to be great! I will merge this change soon.
and, I'm planning make another PR add nomex and environmental_resist support(#2) after CleverRaven#32399.

File: src/iuse_actor.cpp [hunk 1]
```diff
@@ -4376,6 +4376,17 @@ int sew_advanced_actor::use( player &p, item &it, bool, const tripoint & ) const
                ( it.charges >= thread_needed && has_enough[mat_item] );
     };
 
+    const auto get_compare_color = [&](const int before, const int after, const bool higher_is_better) {
+        return before == after ? c_unset : (after > before == higher_is_better ? c_light_green : c_red);
```
Comment (lispcoc) [inline src/iuse_actor.cpp]: gcc warns parentheses error.
```suggestion
        return before == after ? c_unset : ((after > before) == higher_is_better ? c_light_green : c_red);
```

File: src/iuse_actor.cpp [hunk 2]
```diff
@@ -4376,6 +4376,17 @@ int sew_advanced_actor::use( player &p, item &it, bool, const tripoint & ) const
...
+    };
+    const auto get_volume_compare_color = [&](const units::volume before, const units::volume after, const bool higher_is_better) {
+        return before == after ? c_unset : (after > before == higher_is_better ? c_light_green : c_red);
```
Comment (lispcoc) [inline src/iuse_actor.cpp]:
```suggestion
        return before == after ? c_unset : ((after > before) == higher_is_better ? c_light_green : c_red);
```

**Comment by lispcoc** [comment]
build error is fixed. please format code style by execute `make astyle` before merge.

**Comment by Petethegoat** [comment]
Sorry about that. :)

**Pull request merged**